src/sagas.js

```diff
@@ -0,0 +1,78 @@
+import { call, put, takeEvery, takeLatest } from 'redux-saga/effects'
+import {fetchMovies, fetchMovieDetails} from './moviesAPI'
+import { MOVIES_FETCH_SUCCEEDED, SHOW_MOVIE_DETAILS, DETAILS_FETCH_SUCCEEDED, GET_MORE_RESULTS, ERROR } from './redux/actions'
+
+//get movies for search
+function* getMovies(action) {
+   try {
+      const moviesResults = yield call(fetchMovies, action.payload, 1)
+      //chack if not error
+      if (moviesResults.Response === 'True') {
+        //calculate remaining results
+        const remainingResults = moviesResults.totalResults - moviesResults.Search.length
+        //start moviesFetchSucceeded action where page is 1
+        yield put({type: MOVIES_FETCH_SUCCEEDED,
+          payload: {
+            results: moviesResults.Search, searchString: action.payload, remainingResults: remainingResults , page: 1,
+        }
+        });
+      }
+      else {
+        yield put({type: ERROR, payload: moviesResults.Error});
+      }
+   } catch (e) {
+      yield put({type: ERROR, payload: e.message});
+   }
+}
+
+//get more results for prev search string
+function* getMoreResults(action) {
+  try {
+    let remainingResults = action.payload.remainingResults
+    //fetch data only if more results exist
+    if (remainingResults > 0) {
+      //update page number and fetch movies from moviesAPI
+      const page = action.payload.page + 1
+      const moviesResults = yield call(fetchMovies, action.payload.searchString, page)
+      //chack if not error
+      if (moviesResults.Response === 'True') {
+        console.log("MOVIES_FETCH_SUCCEEDED");
+        //calculate remaining results and start moviesFetchSucceeded action
+        remainingResults = remainingResults - moviesResults.Search.length
+        yield put({type: MOVIES_FETCH_SUCCEEDED,
+          payload: {
+            results: moviesResults.Search, searchString: action.payload.searchString, remainingResults: remainingResults , page: page,
+        }
+       });
+      }
+      else {
+        yield put({type: ERROR, payload: moviesResults.Error});
```

Comment:
`moviesResults` is undefined here..

Reply:
it is defined, it returned from server with a "False" value in Response and a Error value